lib/ice_cube/parsers/ical_parser.rb

```diff
@@ -23,6 +25,14 @@ def self.schedule_from_ical(ical_string, options = {})
       Schedule.from_hash data
     end
 
+    def self._parse_in_tzid(value, tzid)
+      time_parser = Time
+      if tzid
+        time_parser = ActiveSupport::TimeZone.new(tzid.split('=')[1]) || Time
```

Comment:
This is actually inconsistent with ice_cube's behavior for `to_ical`. Ice_cube's to_ical uses the `%Z` strftime string which (at least for en_US based locale's, behavior may be different elsewhere) outputs a string like `PDT` which is effectively a utc offset not a timezone (PDT= UTC - 7 hours vs PST = UTC - 8 hours).

But `ActiveSupport::TimeZone.new` requires an actual timezone, either the Olson format timezone such as `America/Los_Angeles` or the other version of `Pacific Time (US & Canada)` (which is some US only format that I don't even know if it's a standard with a name). Furthermore, there's not an ActiveSupport method for going from a timezone offset like PDT to a region's timezone because they're not the same thing.

So, in order for this PR to really make sense in the library, the `to_ical` method should be changed to use the `.time_zone.name` property of ActiveSupport::TimeWithZone instead of the strftime string.

I checked the ical [RFC](https://tools.ietf.org/html/rfc5545#section-3.2.19) and it does not specify timezone identifiers, but says `Implementers may want to use the naming conventions defined in existing time zone specifications such as the public-domain TZ database [TZDB]` and links to the olson [timzone database](http://www.twinsun.com/tz/tz-link.htm)